RCAIDE/Framework/Missions/Conditions/Conditions.py

```diff
@@ -0,0 +1,148 @@
+# RCAIDE/Framework/Missions/Conditions/Conditions.py
+# (c) Copyright 2024 Aerospace Research Community LLC
+#
+# Created: Jul 2024, RCAIDE Team
+
+# ----------------------------------------------------------------------------------------------------------------------
+#  IMPORT
+# ----------------------------------------------------------------------------------------------------------------------
+
+from dataclasses import dataclass
+import unittest
+
+# package imports
+import numpy as np
+
+# ----------------------------------------------------------------------------------------------------------------------
+#  Conditions
+# ----------------------------------------------------------------------------------------------------------------------
+
+
+@dataclass(kw_only=True)
+class Conditions:
+    """
+    A class representing a set of conditions for aerospace simulations.
+
+    This class is designed to hold various attributes and methods related to
+    simulation conditions, allowing for dynamic expansion of rows and packing
+    of data into arrays.
+
+    Attributes
+    ----------
+    name : str
+        The name of the conditions set. Default is 'Conditions'.
+    number_of_rows : int
+        The number of rows in the conditions set. Default is 1.
+    adjustment_from_parent : int
+        An adjustment factor for row calculations. Default is 0.
+
+    Methods
+    -------
+    __post_init__()
+        Initializes the object after creation, expanding rows as needed.
+    expand_rows(rows: int, override: bool = False)
+        Expands or resizes the number of rows in the object and its attributes.
+    pack_array()
+        Packs all numpy arrays in the object into a single flattened array.
+
+    Notes
+    -----
+    - This class is implemented as a dataclass with keyword-only arguments.
+    - It's designed to work with nested Conditions objects and numpy arrays.
+    """
+
+    name: str = 'Conditions'
+
+    number_of_rows: int = 1
+    adjustment_from_parent: int = 0
```

Comment:
I think we discussed this as a group. But I think we should pick a different name. A suggestion, but not one I'm saying needs to be used.

base_row_size_adjustment ?

Reply:
Changed to 'row_size_adjustment'